### PR TITLE
Add b87 keyboard

### PR DESCRIPTION
**Pull request by Jels-kb**
## Description

Added QMK support for the B87, a board produced by winkeyless.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/winkeyless/b87/info.json [hunk 1]
```diff
@@ -0,0 +1,104 @@
+{
+    "keyboard_name": "B87",
+    "url": "",
+    "maintainer": "qmk",
+    "width": 18.25,
+    "height": 6.25,
```
Comment (drashna) [inline keyboards/winkeyless/b87/info.json:6]:
```suggestion
```

File: keyboards/winkeyless/b87/keymaps/via/config.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#define DYNAMIC_KEYMAP_LAYER_COUNT 3
```
Comment (drashna) [inline keyboards/winkeyless/b87/keymaps/via/config.h:17]:
```suggestion
#pragma once

#define DYNAMIC_KEYMAP_LAYER_COUNT 3
```

File: keyboards/winkeyless/b87/rules.mk [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes 	    # Enable keyboard backlight functionality on B7 by default
+UNICODE_ENABLE = no         # Unicode
+AUDIO_ENABLE = no           # Audio output on port C6
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
+WS2812_DRIVER = i2c
```
Comment (drashna) [inline keyboards/winkeyless/b87/rules.mk:30]:
```suggestion
# Bootloader selection
BOOTLOADER = bootloadHID

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = yes 	    # Enable keyboard backlight functionality
AUDIO_ENABLE = no           # Audio output
RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
WS2812_DRIVER = i2c
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/winkeyless/b87/config.h [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+/* Define Matrix size */
+#define MATRIX_ROWS 8
+#define MATRIX_COLS 15
+
+/*Define Matrix Pins */
+#define MATRIX_ROW_PINS {B5, B4, B3, B2, B1, B0, B6, B7}
+#define MATRIX_COL_PINS {A0, A1, A2, A3, A4, A5, C5, C4, C3, C2, D7, C6, C7, A7, A6}
+    
+#define RGBLED_NUM 20  // Number of LEDs
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/winkeyless/b87/config.h:37]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/winkeyless/b87/rules.mk [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+MCU = atmega32a
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/winkeyless/b87/rules.mk:11]:
```suggestion
```

File: keyboards/winkeyless/b87/rules.mk [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes 	    # Enable keyboard backlight functionality on B7 by default
+UNICODE_ENABLE = no         # Unicode
```
Comment (drashna) [inline keyboards/winkeyless/b87/rules.mk:27]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/winkeyless/b87/rules.mk [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover Doesn't work on V-USB
+BACKLIGHT_ENABLE = yes 	    # Enable keyboard backlight functionality on B7 by default
+AUDIO_ENABLE = no           # Audio output on port C6
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
+WS2812_DRIVER = i2c
```
Comment (fauxpark) [inline keyboards/winkeyless/b87/rules.mk:20]:
```suggestion

# Bootloader selection
BOOTLOADER = bootloadhid

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
WS2812_DRIVER = i2c
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**